src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs

```diff
@@ -149,37 +149,26 @@ public static void Run(ExecutionContext executionContext, ContextCallback callba
 
         internal static void RunInternal(ExecutionContext? executionContext, ContextCallback callback, object? state)
         {
-            // Note: ExecutionContext.RunInternal is an extremely hot function and used by every await, ThreadPool execution, etc.
-            // Note: Manual enregistering may be addressed by "Exception Handling Write Through Optimization"
-            //       https://github.com/dotnet/runtime/blob/master/docs/design/features/eh-writethru.md
-
-            // Enregister variables with 0 post-fix so they can be used in registers without EH forcing them to stack
-            // Capture references to Thread Contexts
-            Thread currentThread0 = Thread.CurrentThread;
-            Thread currentThread = currentThread0;
-            ExecutionContext? previousExecutionCtx0 = currentThread0._executionContext;
+            Thread currentThread = Thread.CurrentThread;
+            ExecutionContext? previousExecutionCtx0 = currentThread._executionContext;
```

Comment:
I still had to retain `previousExecutionCtx0` because it is no longer `single def` because of another assignment inside the if block below.

Reply:
probably I should undo deleting the comments for this variable.

Reply:
Are you going to restore the comments?

Reply:
Yes.